assets/js/src/core/components/grid/grid.tsx

```diff
@@ -153,6 +157,48 @@ export const Grid = ({ enableMultipleRowSelection = false, modifiedCells = [], s
     tableProps.columnResizeMode = columnResizeMode
   }
 
+  const [columnSizingInfo, setColumnSizingInfo] = useState<ColumnSizingInfoState>()
+
+  tableProps.onColumnSizingInfoChange = (updater) => {
+    // Update your own state with the new column sizing info
+    const newValue = functionalUpdate(updater, columnSizingInfo)
+
+    if (tableAutoWidth && typeof newValue !== 'undefined' && typeof newValue?.isResizingColumn === 'string') {
+      const column = table.getColumn(newValue.isResizingColumn)
+      const columnWidth = autoColumnRef.current?.clientWidth
+      if (column?.columnDef.meta?.autoWidth === true && typeof columnWidth !== 'undefined') {
+        column.columnDef.size = columnWidth
+        column.columnDef.meta.autoWidth = false
+
+        if (typeof autoColumnRef.current?.clientWidth !== 'undefined') {
+          newValue.startSize = autoColumnRef.current?.clientWidth
+          newValue.columnSizingStart.forEach(columnSizing => {
+            columnSizing[1] = columnWidth
+          })
+        }
+
+        setColumnSizingInfo(newValue)
+        setTableAutoWidth(false)
+        return
+      }
+    }
+
+    setColumnSizingInfo(updater)
+  }
+
+  // validate if only one column has autoWidth set to true
+  if (tableAutoWidth) {
+    let autoWidthColumnFound: boolean = false
+    for (const column of columns) {
+      if (column.meta?.autoWidth === true) {
+        if (autoWidthColumnFound) {
+          throw new Error('Only one column can have autoWidth set to true')
+        }
+        autoWidthColumnFound = true
+      }
+    }
+  }
```

Comment:
We should wrap this in useMemo to only recompute it, when the columns change.
At the moment we do it on every render. Should not have to much effect but for the grid it can not hurt to save some computing time. :)

Reply:
Thanks, implemented this and it seems to work!